jsonrpc/jsonrpc.go

```diff
@@ -0,0 +1,70 @@
+// Package jsonrpc is a minimal JSON-RPC implementation
+package jsonrpc
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type JSONRPCRequest struct {
+	ID      interface{}    `json:"id"`
+	Method  string         `json:"method"`
+	Params  [1]interface{} `json:"params"`
+	Version string         `json:"jsonrpc,omitempty"`
+}
+
+func NewJSONRPCRequest(id interface{}, method string, args interface{}) *JSONRPCRequest {
+	return &JSONRPCRequest{
+		ID:      id,
+		Method:  method,
+		Params:  [1]interface{}{args},
+		Version: "2.0",
+	}
+}
+
+// Send sends the request to URL and returns the general JsonRpcResponse, or an error (note: not the JSONRPCError)
+func (req *JSONRPCRequest) Send(url string) (res *JSONRPCResponse, err error) {
+	buf, err := json.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	rawResp, err := http.Post(url, "application/json", bytes.NewBuffer(buf))
+	if err != nil {
+		return nil, err
+	}
+
+	res = new(JSONRPCResponse)
+	if err := json.NewDecoder(rawResp.Body).Decode(res); err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+type JSONRPCResponse struct {
+	ID      interface{}     `json:"id"`
+	Result  json.RawMessage `json:"result"`
```

Comment:
Result is optional (not present if Error is), so it could be empty.
I guess it doesn't matter as you require every user to check Error before trying to parse the Results.